### PR TITLE
ENH: {arg}sort, {take, put}along_axis

### PR DESCRIPTION
**Pull request by ev-br**
This started as an experiment with `torch.scatter_`; that had a limited success, but at least we have sort and argsort which were needed for tests.

### CODE REVIEW

File: torch_np/_ndarray.py [hunk 1]
```diff
@@ -352,6 +352,12 @@ def clip(self, min, max, out=None):
         result = _impl.clip(tensor, t_min, t_max)
         return _helpers.result_or_out(result, out)
 
+    def repeat(self, repeats, axis=None):
+        t_repeats = asarray(repeats).get()  # XXX: scalar repeats
+        tensor = self._tensor
+        result = torch.repeat_interleave(tensor, t_repeats, axis)
```
[the remark this answers is not in the document]
Comment (ev-br) [reply torch_np/_ndarray.py:358]: Yes. I'm saving this for a next refactor pass, together with a rework of the wrapper machinery.

File: torch_np/_detail/implementations.py [hunk 2]
```diff
@@ -548,3 +536,52 @@ def round(tensor, decimals=0):
...
+    return result
+
+
+# ### sort and partition ###
+
+
+def _sort_helper(tensor, axis, kind, order):
+    if order is not None:
+        # only relevant for structured dtypes; not supported
+        raise NotImplementedError
```
[the remark this answers is not in the document]
Comment (ev-br) [reply torch_np/_detail/implementations.py]: OK. Not sure about a helper function: there might be one for array storage orders ('CFAK'), and this one is likely the only place where this one is used. If it's more common, will refactor into a helper function.

File: torch_np/_detail/implementations.py [hunk 3]
```diff
@@ -548,3 +536,52 @@ def round(tensor, decimals=0):
...
+
+
+def put_along_dim(tensor, t_indices, t_values, axis):
+    (tensor,), axis = _util.axis_none_ravel(tensor, axis=axis)
+    axis = _util.normalize_axis_index(axis, tensor.ndim)
+
+    result = tensor.clone()
+    t_indices, t_values = torch.broadcast_tensors(t_indices, t_values)
+    t_values = _util.cast_if_needed(t_values, result.dtype)
+    result.scatter_(axis, t_indices, t_values)
```
[the remark this answers is not in the document]
Comment (ev-br) [reply torch_np/_detail/implementations.py]: Was using an in-place torch function for an in-place numpy function. That said, it's using `torch.scatter` now.

File: torch_np/_detail/implementations.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lezcano) [reply torch_np/_detail/implementations.py]: Fine. Let's just add a more informative error then.

[the remark this answers is not in the document]
Comment (lezcano) [reply torch_np/_detail/implementations.py]: ah, I had missed the update. Thank you!

File: torch_np/_ndarray.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lezcano) [reply torch_np/_ndarray.py:358]: I filed an issue so that we remember https://github.com/Quansight-Labs/numpy_pytorch_interop/issues/60

**Review by lezcano — APPROVED** [review]

**Pull request merged**